pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentDispatcherSingleActiveConsumer.java

```diff
@@ -174,9 +210,18 @@ public synchronized void redeliverUnacknowledgedMessages(Consumer consumer) {
                     consumer);
             return;
         }
+
+        if (readOnActiveConsumerTask != null && (!readOnActiveConsumerTask.isDone()
+                || readOnActiveConsumerTask.isCancelled())) {
+            log.info("[{}] Ignoring reDeliverUnAcknowledgedMessages: consumer is waiting for cursor to be rewinded",
```

Comment:
can we add `name` to log and also at line#209 which has topic-name in it and it will help for debugging.

Reply:
Fixed.